src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/AvaloniaXamlIlLanguageParseIntrinsics.cs

```diff
@@ -291,6 +291,26 @@ public static bool TryConvert(AstTransformationContext context, IXamlAstValueNod
                 return true;
             }
 
+            if (type.Equals(types.ThemeVariant))
+            {
+                var variantText = text.Trim();
+                var foundConstProperty = types.ThemeVariant.Properties.FirstOrDefault(p =>
+                    p.Name == variantText && p.PropertyType == types.ThemeVariant);
+                var themeVariantTypeRef = new XamlAstClrTypeReference(node, types.ThemeVariant, false);
+                if (foundConstProperty is not null)
+                {
+                    result = new XamlStaticExtensionNode(new XamlAstObjectNode(node, node.Type), themeVariantTypeRef, foundConstProperty.Name);
+                    return true;
+                }
+
+                result = new XamlAstNewClrObjectNode(node, themeVariantTypeRef, types.ThemeVariantConstructor,
```

Comment:
As discussed separately, auto-constructing an incomplete ThemeVariant instance is a bad idea